cmd/oras/internal/errors/errors.go

```diff
@@ -62,10 +62,18 @@ func CheckArgs(checker func(args []string) (bool, string), Usage string) cobra.P
 
 // NewErrEmptyTagOrDigest creates a new error based on the reference string.
 func NewErrEmptyTagOrDigest(ref registry.Reference) error {
-	return NewErrEmptyTagOrDigestStr(ref.String())
+	return NewErrEmptyTagOrDigestStr(ref.String(), nil, false)
 }
 
 // NewErrEmptyTagOrDigestStr creates a new error based on the reference string.
-func NewErrEmptyTagOrDigestStr(ref string) error {
-	return fmt.Errorf("%q: no tag or digest when expecting <name:tag|name@digest>", ref)
+func NewErrEmptyTagOrDigestStr(ref string, cmd *cobra.Command, needsTag bool) error {
+	form := `"<name>@<digest>"`
+	if needsTag {
+		form = fmt.Sprintf(`"<name>:<tag>" or %s`, form)
+	}
+	return &Error{
+		Err:            fmt.Errorf(`"%s": no tag or digest specified`, ref),
```

Comment:
For `blob` operation need to get a error like `fmt.Errorf(`"%s": no digest specified`, ref),`, might need another form variable

Reply:
refined with a new variable.